### ISSUE TITLE
Clone throws exception if the repository name is misspelled

### ISSUE THREAD

**Issue by mdiep** [Carthage/Carthage#1355]
* `carthage version`: 0.16.2

**Cartfile**
```
github "groue/GRDB" "master"
```

**Carthage Output**
```
$ carthage update --no-build --use-submodules
*** Cloning GRDB
2016-06-18 20:41:38.225 carthage[22687:1237651] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'working directory doesn't exist.'
*** First throw call stack:
(
…
```

**Comment by drekka** [comment]
Just hit this one after blowing away cache directory to fix issue with commits in local repo not being picked up. Am going to try checking out without a branch specifier first. 

+1 to fix this and the other bug urgently. :-)

**Comment by mdiep** [comment]
Would you be interested in trying to fix it? 😄

**Comment by drekka** [comment]
I had a look at the codebase, but unfortunately I don't know any of the APIs you guys are using and I'm flat out with some other projects, or I would :-(

**Comment by mdiep** [comment]
Fixed by #1451.
